### PR TITLE
CSHARP-3236: Verify that nested Contains works in LINQ3.

### PR DESCRIPTION
**Pull request by rstam**

### CODE REVIEW

File: tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp3326Tests.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                    Id = p.Id,
+                    Comments = p.Comments.Where(c => c.Text.Contains("test"))
+                });
+
+            var stages = Translate(collection, queryable);
+            AssertStages(
+                stages,
+                "{ $project : { Id : '$_id', Comments : { $filter : { input : '$Comments', as : 'c', cond : { $gte : [{ $indexOfCP : ['$$c.Text', 'test'] }, 0] } } }, _id : 0 } }");
+
+            var result = queryable.Single();
```
Comment (JamesKovacs) [inline tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp3326Tests.cs:43]: We should assert the returned result here.

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Missing assertion in test. Given that these were omissions/bugs reported in LINQ2 that now work in LINQ3, I think we should incorporate these tests into our suite to prove that LINQ3 now handles these cases.

File: tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp3326Tests.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+*/
+
+using System.Collections.Generic;
+using System.Linq;
+using MongoDB.Driver.Linq;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Linq.Linq3ImplementationTests.Jira
+{
+    public class CSharp3236Tests : Linq3IntegrationTest
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp3326Tests.cs]: The file name was wrong.

I've renamed it.

File: tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp3326Tests.cs (hunk 1 above)
Comment (rstam) [reply tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp3326Tests.cs]: Yes! An inadvertent omission.

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**